scripts/eachdist.py

```diff
@@ -0,0 +1,508 @@
+
+        This option optionally takes a format string to apply to each path. The
+        available replacements are the ones that would be available for `format`
+        if `--all` was not specified.
+
+        Default ALLFORMAT if this flag is specified: `%(const)s`.
+        """
+        ),
+    )
+    excparser.add_argument(
+        "--allsep",
+        help=cleandoc(
+            """Separator string for the strings resulting from `--all`.
+        Only valid if `--all` is specified.
+        """
+        ),
+    )
+    excparser.add_argument(
+        "--allowexitcode",
+        type=int,
+        action="append",
+        default=[0],
+        help=cleandoc(
+            """The given command exit code is treated as success and does not abort execution.
+        Can be specified multiple times.
+        """
+        ),
+    )
+    excparser.add_argument(
+        "--mode",
+        "-m",
+        default="DEFAULT",
+        help=cleandoc(
+            """Section of config file to use for target selection configuration.
+        See description of exec for available options."""
+        ),
+    )
+
+    instparser = subparsers.add_parser(
+        "install", help="Install all distributions."
+    )
+
+    def setup_instparser(instparser):
+        instparser.set_defaults(func=install_args)
+        instparser.add_argument(
+            "pipargs", nargs=argparse.REMAINDER, help=extraargs_help("pip")
+        )
+
+    setup_instparser(instparser)
+    instparser.add_argument("--editable", "-e", action="store_true")
+    instparser.add_argument("--with-dev-deps", action="store_true")
+    instparser.add_argument("--eager-upgrades", action="store_true")
+
+    devparser = subparsers.add_parser(
+        "develop",
+        help="Install all distributions editable + dev dependencies.",
+    )
+    setup_instparser(devparser)
+    devparser.set_defaults(
+        editable=True, with_dev_deps=True, eager_upgrades=True
+    )
+
+    lintparser = subparsers.add_parser(
+        "lint", help="Lint everything, autofixing if possible."
+    )
+    lintparser.add_argument("--check-only", action="store_true")
+    lintparser.set_defaults(func=lint_args)
+
+    testparser = subparsers.add_parser(
+        "test",
+        help="Test everything (run pytest yourself for more complex operations).",
+    )
+    testparser.set_defaults(func=test_args)
+    testparser.add_argument(
+        "pytestargs", nargs=argparse.REMAINDER, help=extraargs_help("pytest")
+    )
+
+    return parser.parse_args(args)
+
+
+def find_projectroot(search_start=Path(".")):
+    root = search_start.resolve()
+    for root in chain((root,), root.parents):
+        if any((root / marker).exists() for marker in (".git", "tox.ini")):
+            return root
+    return None
+
+
+def find_targets_unordered(rootpath):
+    for subdir in rootpath.iterdir():
+        if not subdir.is_dir():
+            continue
+        if subdir.name.startswith(".") or subdir.name.startswith("venv"):
```

Comment:
Using `gitignore` or similar seems like a better long term solution here.

Reply:
True, actually since Python already supports globbing (and it's already used in that script), that should be a relatively simple change.